src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -254,25 +250,24 @@ otError RadioSpinel::CheckSpinelVersion(void)
     return error;
 }
 
-otError RadioSpinel::CheckCapabilities(bool &aIsRcp)
+template <typename InterfaceType, typename ProcessContextType>
+bool RadioSpinel<InterfaceType, ProcessContextType>::IsRcp(void)
 {
-    otError        error = OT_ERROR_NONE;
     uint8_t        capsBuffer[kCapsBufferSize];
     const uint8_t *capsData         = capsBuffer;
     spinel_size_t  capsLength       = sizeof(capsBuffer);
     bool           supportsRawRadio = false;
+    bool           isRcp            = false;
 
-    SuccessOrExit(error = Get(SPINEL_PROP_CAPS, SPINEL_DATATYPE_DATA_S, capsBuffer, &capsLength));
-
-    aIsRcp = false;
+    SuccessOrDie(Get(SPINEL_PROP_CAPS, SPINEL_DATATYPE_DATA_S, capsBuffer, &capsLength));
 
     while (capsLength > 0)
     {
         unsigned int   capability;
         spinel_ssize_t unpacked;
 
         unpacked = spinel_datatype_unpack(capsData, capsLength, SPINEL_DATATYPE_UINT_PACKED_S, &capability);
-        VerifyOrExit(unpacked > 0, error = OT_ERROR_FAILED);
+        VerifyOrDie(unpacked > 0, OT_EXIT_RADIO_SPINEL_INCOMPATIBLE);
 
         if (capability == SPINEL_CAP_OPENTHREAD_LOG_METADATA)
```

Comment:
Delete line 272 - 280? `SPINEL_CAP_OPENTHREAD_LOG_METADATA` and `SPINEL_CAP_MAC_RAW` are not related to IsRcp check.

Reply:
Moved to `CheckRadioCapabilities`.